jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -30,17 +32,22 @@ class JobRunExecutorActor(
     driverHolder:               MarathonSchedulerDriverHolder,
     clock:                      Clock,
     val behavior:               Behavior
-) extends Actor with Stash with ActorLogging with ActorBehavior {
+)(implicit scheduler: Scheduler) extends Actor with Stash with ActorLogging with ActorBehavior {
   import JobRunExecutorActor._
   import JobRunPersistenceActor._
   import TaskStates._
+  import context.dispatcher
+
+  private[impl] var startingDeadlineTimer: Option[Cancellable] = None
 
   lazy val persistenceActor = persistenceActorRefFactory(run.id, context)
   var jobRun: JobRun = run
 
   val runSpecId = jobRun.id.toPathId
 
   override def preStart(): Unit = {
+    scheduleTimeout()
```

Comment:
when we agreed to change the name to simply `scheduleTimeout` it was with the passing of startingDeadline such as `scheduleTimeout(startingDeadline)`.   now the deadline is in the job run so perhaps it is `scheduleTimeout(jobRun.startingDeadline)`.    I do not like the abbreviated name without the context of what timeout we are scheduling.

Reply:
good point, I'll change the name